internalshared/configutil/listener.go

```diff
@@ -279,27 +292,28 @@ func (l *Listener) parseChrootNamespaceSettings() error {
 // The state of the listener will be modified.
 func (l *Listener) parseType(fallback string) error {
 	switch {
-	case l.Type != "":
+	case l.Type.String() != "":
 	case fallback != "":
 	default:
 		return errors.New("listener type must be specified")
 	}
 
 	// Use type if available, otherwise fall back.
-	result := l.Type
-	if result == "" {
-		result = fallback
+	rawType := l.Type
+	if rawType == "" {
```

Comment:
Why is it that `""` doesn't need to be cast to **listenerType** or compared to `rawType.String()` like is being done on line 295 above?